test/python/circuit/test_instructions.py

```diff
@@ -84,6 +84,23 @@ def test_instructions_equal_with_parameters(self):
         self.assertNotEqual(Instruction('u', 1, 0, [0.3, phi, 0.4]),
                             Instruction('u', 1, 0, [theta, phi, 0.5]))
 
+    def test_instructions_soft_compare(self):
+        """Test soft comparison between instructions."""
+        theta = Parameter('theta')
+        phi = Parameter('phi')
+
+        # Verify that we are insensitive when there are parameters.
+        self.assertTrue(Instruction('u', 1, 0, [0.3, phi, 0.4]).soft_compare(
+            Instruction('u', 1, 0, [theta, phi, 0.4])))
+
+        # Verify that normal equality still holds.
+        self.assertTrue(Instruction('u', 1, 0, [0.4, 0.5]).soft_compare(
+            Instruction('u', 1, 0, [0.4, 0.5])))
+
+        # Test that when names differ we get False.
+        self.assertFalse(Instruction('u', 1, 0, [0.4, phi]).soft_compare(
+            Instruction('v', 1, 0, [theta, phi])))
+
```

Comment:
Please add a test that invokes the cutoff precision check in soft compare

Reply:
Sure but I think we may want to be a bit careful here. On my machine this
```
aa = np.array([1., 0.40000+1e-6])
bb = np.array([1., 0.40000])
np.allclose(aa, bb, atol=1e-10)
```
 gives me `True`. But this
```
aa = np.array([1., 0.40000+1e-5])
bb = np.array([1., 0.40000])
np.allclose(aa, bb, atol=1e-10)
```
gives me `False`. This does not quite line up with `_CUTOFF_PRECISION = 1E-10`.

Reply:
Can we make it `+1e-20` for True and `+0.01` for false?